homeassistant/helpers/device.py

```diff
@@ -39,7 +43,11 @@ def async_device_info_to_link_from_device_id(
     hass: HomeAssistant,
     device_id: str | None,
 ) -> dr.DeviceInfo | None:
-    """DeviceInfo with information to link a device to a configuration entry in the link category from a device id."""
+    """DeviceInfo with information to link a device from a device id.
+
+    DeviceInfo will only return the information to categorize the config
```

Comment:
We shouldn't talk about config entry here, I think. We're returning a device entry and it's that item that belongs to a link category here.